### PR TITLE
Feature/eurolinux support

### PR DESCRIPTION
**Pull request by AlexBaranowski**
Hi,

This PR enables support for EuroLinux distro. 

EuroLinux is mostly a clone of RHEL. I made these changes based on what is enabled/made for CentOS.
I backported changes to our version of cloud-init for EL7 (Enterprise Linux/EuroLinux), and it works like a charm. It also works on EL8.

I changed .gitignore a little bit - coverage makes a lot of files like `.coverage.$(hostname).XXX.YYYYYYY` during a run, so they can be added by mistake.  cloud-init*src.rpm and rpm files are also ignored.

**Any suggestions/comments are incredibly welcome.**

### CODE REVIEW

File: cloudinit/sources/DataSourceRbxCloud.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def _sub_arp(cmd):
 
 def gratuitous_arp(items, distro):
     source_param = '-S'
-    if distro.name in ['fedora', 'centos', 'rhel']:
+    if distro.name in ['fedora', 'centos', 'rhel', 'eurolinux']:
```
[the remark this answers is not in the document]
Comment (AlexBaranowski) [reply cloudinit/sources/DataSourceRbxCloud.py:56]: Yeah, this should be removed.

File: tests/cloud_tests/setup_image.py [hunk 2]
```diff
@@ -180,7 +180,7 @@ def enable_repo(args, image):
     if os_family == 'debian':
         cmd = ('echo "{}" >> "/etc/apt/sources.list" '.format(args.repo) +
                '&& apt-get update')
-    elif os_family == 'centos':
+    elif os_family == 'centos' or os_family == 'eurolinux':
```
[the remark this answers is not in the document]
Comment (AlexBaranowski) [reply tests/cloud_tests/setup_image.py]: Thanks for your input - it's now removed.

File: cloudinit/distros/eurolinux.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# This file is part of cloud-init. See LICENSE file for license information.
+
+from cloudinit.distros import rhel
+from cloudinit import log as logging
+
+LOG = logging.getLogger(__name__)
```
Comment (OddBloke) [inline cloudinit/distros/eurolinux.py:6]: No need to set up `LOG` as it's unused.

File: cloudinit/tests/test_util.py [hunk 4]
```diff
@@ -454,6 +475,14 @@ def test_get_linux_freebsd(self, m_parse_redhat_release,
         dist = util.get_linux_distro()
         self.assertEqual(('freebsd', '12.0-RELEASE-p10', ''), dist)
 
+    @mock.patch('cloudinit.util.load_file')
+    def test_get_linux_eurolinux7_redhat_release(self, m_os_release, m_exists):
+        """Verify the correct release info on CentOS 7 without os-release."""
```
Comment (OddBloke) [inline cloudinit/tests/test_util.py:480]:
```suggestion
        """Verify the correct release info on EuroLinux 7 without os-release."""
```

File: cloudinit/distros/eurolinux.py (hunk 3 above)
Comment (AlexBaranowski) [reply cloudinit/distros/eurolinux.py]: Fixed

File: cloudinit/tests/test_util.py (hunk 4 above)
Comment (AlexBaranowski) [reply cloudinit/tests/test_util.py]: Fixed

**Comment by AlexBaranowski** [comment]
When it comes to following files
```
cloudinit/config/cc_ntp.py:27:distros = ['centos', 'debian', 'fedora', 'opensuse', 'rhel', 'sles', 'ubuntu']
cloudinit/config/cc_resolv_conf.py:58:distros = ['fedora', 'opensuse', 'rhel', 'sles']
cloudinit/sources/DataSourceRbxCloud.py:56:    if distro.name in ['fedora', 'centos', 'rhel']:
```
cc_ntp - requires own ntp pool that we don't have.  
cloudinit/config/cc_resolv_conf.py - as long as CentOS is not listed I shoudn't overstep :stuck_out_tongue_closed_eyes:.  
cloudinit/sources/DataSourceRbxCloud.py - they don't offer our system, so adding EuroLinux will be misleading.

**Comment by OddBloke** [comment]
> `cloudinit/config/cc_ntp.py:27:distros = ['centos', 'debian', 'fedora', 'opensuse', 'rhel', 'sles', 'ubuntu']`
> cc_ntp - requires own ntp pool that we don't have.

This isn't the case. The line in question is used to determine whether cloud-init will execute this module on a distro at all. If you don't add "eurolinux" here, then any EuroLinux users who specify NTP configuration to cloud-init will not have it applied to the system.

> `cloudinit/config/cc_resolv_conf.py:58:distros = ['fedora', 'opensuse', 'rhel', 'sles']`
> cloudinit/config/cc_resolv_conf.py - as long as CentOS is not listed I shoudn't overstep stuck_out_tongue_closed_eyes.

That's fair enough; I don't know enough about the subtleties here to know what's right. :)

> `cloudinit/sources/DataSourceRbxCloud.py:56:    if distro.name in ['fedora', 'centos', 'rhel']:`
> cloudinit/sources/DataSourceRbxCloud.py - they don't offer our system, so adding EuroLinux will be misleading.

Users can bring their own images to most clouds; if you don't add "eurolinux" to this list, then any users who attempt to use EuroLinux on Rootbox/Hyperone platforms will (a) see a WARNING in their cloud-init log file, and (b) will experience whatever issue this call is intended to address.

File: cloudinit/tests/test_util.py [hunk 5]
```diff
@@ -107,12 +107,33 @@
     REDHAT_SUPPORT_PRODUCT_VERSION="7.5"
 """)
 
+
+OS_RELEASE_EUROLINUX = dedent("""\
```
Comment (OddBloke) [inline cloudinit/tests/test_util.py:111]: I think we're missing a test that actually uses this.

**Comment by AlexBaranowski** [comment]
Sorry for such a slow response. I'm grateful for the review. I add test to test_utils that uses OS_RELEASE_EUROLINUX variable. I added EuroLinux to DataSourceRbxCloud - as it was pointed, there is probably the possibility to bring own image I didn't contact this cloud provider to check it. Still, it won't hurt to put EuroLinux here.

When it comes to `cc_ntp.py`  EuroLinux will use empty pool_distro var (same as Alpine), so it will use 1.ntp.org 2.ntp.org ... etc. as NTP server. I put the NTP server pool on our wishlist, but because none of our customers asked us about it, so it has low priority.

TIA :smiling_face_with_three_hearts:

File: doc/rtd/topics/availability.rst [hunk 6]
```diff
@@ -25,6 +25,7 @@ and OpenBSD:
 - Gentoo Linux
 - NetBSD
 - OpenBSD
+- EuroLinux
```
Comment (OddBloke) [inline doc/rtd/topics/availability.rst:28]: This list is alphabetised.

**Comment by OddBloke** [comment]
Hey Alex, apologies for missing your last update! This is waiting on us for review, so I'm reopening it.

**Review by OddBloke — COMMENTED** [review]
Just noticed one minor thing in my re-review just now; otherwise the code looks good.

Before we land this, I'd like confirmation that you've tested the latest iteration of this code in Eurolinux and that it's working as expected. Would you be able to spin up an instance with this code included, and then run `cloud-init collect-logs` and attach the tarball it produces to this PR, for validation?

Thanks!

**Comment by AlexBaranowski** [comment]
@OddBloke Sorry for delay, but we are still making new version of EL (7.8 -> 7.9). This version have rebased version of cloud-init. I will povide logs ASAP when new system will be ready. Once more sorry for delay.

**Comment by igalic** [comment]
@AlexBaranowski this pr is still reporting .gitignore as conflicting

**Pull request closed without merging**